performance_manager/lib/l1_full_trip_events.py

```diff
@@ -200,50 +180,51 @@ def pull_and_transform(
             isouter=True,
         )
         .where(
-            (TripUpdateEvents.stop_sequence.isnot(None))
-            & (TripUpdateEvents.timestamp_start > min_ts_process)
-            & (TripUpdateEvents.timestamp_start < max_ts_process)
+            (TripUpdateEvents.updated_on > last_full_event_update_query)
+            & (TripUpdateEvents.stop_sequence.isnot(None))
             & (TripUpdateEvents.stop_id.isnot(None))
             & (TripUpdateEvents.direction_id.isnot(None))
             & (TripUpdateEvents.route_id.isnot(None))
             & (TripUpdateEvents.start_date.isnot(None))
             & (TripUpdateEvents.start_time.isnot(None))
             & (TripUpdateEvents.vehicle_id.isnot(None))
-            & (FullTripEvents.fk_tu_stopped_event.is_(None))
         )
         .cte(name="stop_tu")
     )
 
-    merged_query = (
+    first_join_cte = (
         sa.select(
-            move_vp_cte.c.fk_vp_moving_event,
-            stop_tu_cte.c.fk_tu_stopped_event,
-            stop_vp_cte.c.fk_vp_stopped_event,
+            move_vp_cte.c.fk_vp_moving_event.label("fk_vp_moving_event"),
+            stop_tu_cte.c.fk_tu_stopped_event.label("fk_tu_stopped_event"),
             sa.func.coalesce(
                 move_vp_cte.c.stop_sequence,
-                stop_vp_cte.c.stop_sequence,
+                stop_tu_cte.c.stop_sequence,
             ).label("stop_sequence"),
             sa.func.coalesce(
-                move_vp_cte.c.parent_station, stop_vp_cte.c.parent_station
+                move_vp_cte.c.parent_station,
+                stop_tu_cte.c.parent_station,
             ).label("parent_station"),
             sa.func.coalesce(
                 move_vp_cte.c.direction_id,
-                stop_vp_cte.c.direction_id,
+                stop_tu_cte.c.direction_id,
             ).label("direction_id"),
             sa.func.coalesce(
-                move_vp_cte.c.route_id, stop_vp_cte.c.route_id
+                move_vp_cte.c.route_id,
+                stop_tu_cte.c.route_id,
             ).label("route_id"),
             sa.func.coalesce(
-                move_vp_cte.c.start_date, stop_vp_cte.c.start_date
+                move_vp_cte.c.start_date,
+                stop_tu_cte.c.start_date,
             ).label("start_date"),
             sa.func.coalesce(
-                move_vp_cte.c.start_time, stop_vp_cte.c.start_time
+                move_vp_cte.c.start_time,
+                stop_tu_cte.c.start_time,
             ).label("start_time"),
             sa.func.coalesce(
-                move_vp_cte.c.vehicle_id, stop_vp_cte.c.vehicle_id
+                move_vp_cte.c.vehicle_id,
+                stop_tu_cte.c.vehicle_id,
             ).label("vehicle_id"),
-        )
-        .join(
+        ).join(
```

Comment:
I made the `JOIN`s between the 3 CTE's `FULL` operations, necessitating an intermediate CTE for the first of 2 `JOIN FULL` operations.